charger/ocpp.go

```diff
@@ -336,30 +349,34 @@ func (c *OCPP) setChargingProfile(connectorid int, profile *types.ChargingProfil
 
 // setPeriod sets a single charging schedule period with given current and phases
 func (c *OCPP) setPeriod(current float64, phases int) error {
-	period := types.NewChargingSchedulePeriod(0, current)
-
 	c.log.TRACE.Printf("current phases: %d, current current: %f", phases, current)
-	if phases > 0 {
-		period.NumberPhases = &phases
-	}
 
-	// connectorID: 0 - profile will be applied to all connectors
-	err := c.setChargingProfile(0, getMaxChargingProfile(period))
-	if err != nil {
-		err = fmt.Errorf("failed to set charging profile: %w", err)
+	enabled, err := c.Enabled()
+	if err == nil {
+		if enabled {
+			if err = c.setChargingProfile(c.connector, getTxChargingProfile(current, phases)); err != nil {
+				err = fmt.Errorf("failed to set charging profile: %w", err)
+			}
+		} else {
+			c.log.TRACE.Printf("no running transaction, ommitting request to change charging parameters")
+		}
 	}
 
 	return err
 }
 
-func getMaxChargingProfile(period types.ChargingSchedulePeriod) *types.ChargingProfile {
+func getTxChargingProfile(current float64, phases int) *types.ChargingProfile {
+	period := types.NewChargingSchedulePeriod(0, current)
+	if phases != 0 {
+		period.NumberPhases = &phases
+	}
+
 	return &types.ChargingProfile{
 		ChargingProfileId:      1,
-		StackLevel:             1,
-		ChargingProfilePurpose: types.ChargingProfilePurposeChargePointMaxProfile,
-		ChargingProfileKind:    types.ChargingProfileKindAbsolute,
+		StackLevel:             0,
+		ChargingProfilePurpose: types.ChargingProfilePurposeTxProfile,
+		ChargingProfileKind:    types.ChargingProfileKindRelative,
```

Comment:
Could you also explain what the meaning of "relative" is here?

Reply:
Absolute start time has to be stated explicitly. Relative is counted from the start of transaction. 0 = start of transaction.